GoogleForCreators/web-stories-wp
# FTUE: Animated Skeleton for Help Center

## Context
Finishing up FTUE presentational components and animated skeleton work

## Summary
- Adds Popup & Companion components to tie everything together
- Initial `useHelpCenter` component logic & tests
- Tightens up transitions
- Fixes delay and race conditions with react-transition-groups `Transition` component

## Relevant Technical Choices
Kept all state tied to a single reducer to keep renders lean between hooks.

## To-do
In follow up PRs we need to:
- Add autoplaying mp4s to each quick tip
- add read/unread notification logic and BE persistence
- Smart open/close behavior based off of notifications
- Smart navigation flow based off of read/unread tips

## User-facing changes
Adds initial animated skeleton for FTUE behind enableQuickTips feature flag:

https://user-images.githubusercontent.com/35983235/106527735-fb644a00-64a4-11eb-8741-3e2c9ad65558.mp4


## Testing Instructions
-> experiments -> enableQuickTips
-> Editor -> open the help center and navigate through quick tips -> should have animating  transitions between quicktips and navigation buttons should work. Tips should not have videos & notifications will be done in separate PRs (see todos) 


### QA






### UAT







This PR can be tested by following these steps:

1. experiments -> enableQuickTips
2. Editor -> Click through Tips and see that transitions are satifactory

## Reviews

### Does this PR have a security-related impact?



### Does this PR change what data or activity we track or use?



### Does this PR have a legal-related impact?



## Checklist




---


Fixes #5884
